State that clang binary should be available in the PATH

# System info: #
OS: Windows 10
- Sublime Text version: 3.1.1
- Clang version: clang version 7.0.2 (bundled in android ndk version r18b)

I am using EasyClangComplete plugin, and it does not seem to be doing anything.

I am using the Android ndk version 18b which is bundled with clang, and am developing on Windows. My clang compiler is not located in any default location on Windows. But since the docs say nothing about pointing to your clang location, I will assume it figures this out somehow.

My project is using Cmake, and a CMakeLists.txt is located in the project root. So after extensively reading the docs, its seems like I am set. Things should be working out of the box....which is not the case.

Have I potentially misinterpreted the instructions?

Here is the VERBOSE log
 

<details>
<summary>Verbose log</summary>

```
[ECC:DEBUG]:[EasyClangComplete.py]:[config_updated]:[Thread-6]: updated config: None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
Package Control: Unable to remove old directory JavaScript Completions - deferring until next start
Package Control: Skipping automatic upgrade, last run at 2018-11-30 15:10:18, next run at 2018-11-30 16:10:18 or after
[ECC:DEBUG]:[EasyClangComplete.py]:[on_close]:[MainThread]: closing view 25
[ECC:DEBUG]:[settings_manager.py]:[clear_for_view]:[MainThread]: clearing settings for view: 25
[ECC:DEBUG]:[view_config.py]:[clear_for_view]:[Thread-6]: trying to clear config for view: 25
[ECC:DEBUG]:[EasyClangComplete.py]:[config_removed]:[Thread-6]: removed config for id: 25
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[MainThread]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-5]: view file_name is None
[ECC:DEBUG]:[settings_manager.py]:[settings_for_view]:[Dummy-5]: no settings for view 40. Reinitializing.
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Overriding settings by project ones if needed:
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Valid prefixes: ['ecc_', 'easy_clang_complete_']
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Reading settings...
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Settings sucessfully read...
[ECC:INFO]:[tools.py]:[get_clang_version_str]:[Dummy-5]: Getting version from command: `clang++ -v`
[ECC:DEBUG]:[tools.py]:[run_command]:[Dummy-5]: executable file not found executing: ['clang++', '-v']
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 611, in on_activated_async
    callback.on_activated_async(v)
  File "C:\Users\[redacted_user]\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 274, in on_activated_async
    settings = EasyClangComplete.settings_manager.settings_for_view(view)
  File "C:\Users\[redacted_user]\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_manager.py", line 55, in settings_for_view
    self.__init_for_view(view)
  File "C:\Users\[redacted_user]\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_manager.py", line 103, in __init_for_view
    self.__settings_dict[view_id].update_from_view(view)
  File "C:\Users\[redacted_user]\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_storage.py", line 130, in update_from_view
    self.__update_wildcard_values(view)
  File "C:\Users\[redacted_user]\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_storage.py", line 339, in __update_wildcard_values
    version_str = Tools.get_clang_version_str(self.clang_binary)
  File "C:\Users\[redacted_user]\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\tools.py", line 726, in get_clang_version_str
    if "Apple" in output_text:
TypeError: argument of type 'NoneType' is not iterable
```

</details>




<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/66896739-state-that-clang-binary-should-be-available-in-the-path?utm_campaign=plugin&utm_content=tracker%2F32731927&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32731927&utm_medium=issues&utm_source=github).
</bountysource-plugin>

Hello, thanks for your response. I have set the path to my clang compiler, and it is still not working.

I should note that I am using the Android NDK, which uses clang. I am building on windows but targeting arm.

> 
> [ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Overriding settings by project ones if needed:
> [ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Valid prefixes: ['ecc_', 'easy_clang_complete_']
> [ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Reading settings...
> [ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[Dummy-5]: Settings sucessfully read...
> [ECC:INFO]:[tools.py]:[get_clang_version_str]:[Dummy-5]: Getting version from command: `C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe -v`
> [ECC:DEBUG]:[settings_storage.py]:[__populate_common_flags]:[Dummy-5]: populating common_flags with current variables.
> [ECC:DEBUG]:[settings_storage.py]:[__replace_wildcard_if_needed]:[Dummy-5]: populated '-I$project_base_path/src' to '-IC:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots/src'
> [ECC:DEBUG]:[settings_storage.py]:[__replace_wildcard_if_needed]:[Dummy-5]: populated '-I/usr/lib/clang/$clang_version/include' to '-I/usr/lib/clang/7.0.2/include'
> [ECC:DEBUG]:[settings_storage.py]:[__replace_wildcard_if_needed]:[Dummy-5]: populated '~/some_folder/*' to 'C:\Users\sfield/some_folder/*'
> [ECC:DEBUG]:[settings_storage.py]:[__replace_wildcard_if_needed]:[Dummy-5]: populated '$project_base_path/some/project/path/*' to 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots/some/project/path/*'
> [ECC:DEBUG]:[settings_manager.py]:[__init_for_view]:[Dummy-5]: settings initialized for view: 106
> [ECC:DEBUG]:[EasyClangComplete.py]:[on_activated_async]:[Dummy-5]: on_activated_async view id 106
> [ECC:DEBUG]:[view_config.py]:[load_for_view]:[Thread-9]: generate new config for path: 106
> [ECC:INFO]:[view_config.py]:[__init_completer]:[Thread-9]: init completer based on libclang
> [ECC:DEBUG]:[lib_complete.py]:[__init__]:[Thread-9]: using bundled cindex: EasyClangComplete.plugin.clang.cindex50
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: platform: ('64bit', 'WindowsPE')
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: python version: 3.3.6
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: we are on 'Windows'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: user provided libclang_path: <some_path_here>
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: searching for: 'libclang.dll'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: libclang search output = 'C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang'
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: real output: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: architecture: ('64bit', 'WindowsPE')
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: folder to search: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang\..
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: Checking path: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\libclang.dll
> [ECC:WARNING]:[utils.py]:[find_libclang_dir]:[Thread-9]: Clang could not find 'libclang.dll'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: searching for: 'clang.dll'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: libclang search output = 'C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang'
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: real output: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: architecture: ('64bit', 'WindowsPE')
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: folder to search: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang\..
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: Checking path: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang.dll
> [ECC:WARNING]:[utils.py]:[find_libclang_dir]:[Thread-9]: Clang could not find 'clang.dll'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: searching for: 'libclang.lib'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: libclang search output = 'C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang'
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: real output: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: architecture: ('64bit', 'WindowsPE')
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: folder to search: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang\..
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: Checking path: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\libclang.lib
> [ECC:WARNING]:[utils.py]:[find_libclang_dir]:[Thread-9]: Clang could not find 'libclang.lib'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: searching for: 'clang.lib'
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: libclang search output = 'C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang'
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: real output: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: architecture: ('64bit', 'WindowsPE')
> [ECC:DEBUG]:[utils.py]:[dir_from_output]:[Thread-9]: folder to search: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang\..
> [ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-9]: Checking path: C:\AndroidTools\android-ndk-r18b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang.lib
> [ECC:WARNING]:[utils.py]:[find_libclang_dir]:[Thread-9]: Clang could not find 'clang.lib'
> [ECC:ERROR]:[utils.py]:[find_libclang_dir]:[Thread-9]: no libclang found at all
> [ECC:ERROR]:[lib_complete.py]:[__init__]:[Thread-9]: error: bad argument type for built-in operation
> [ECC:ERROR]:[view_config.py]:[__init_completer]:[Thread-9]: cannot initialize completer with libclang.
> [ECC:INFO]:[view_config.py]:[__init_completer]:[Thread-9]: falling back to using clang in a subprocess.
> [ECC:INFO]:[view_config.py]:[__init_completer]:[Thread-9]: init completer based on clang from cmd
> [ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-9]: [cmake]:[get]: for file C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo\PBScreen.cpp
> [ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-9]: [cmake]:[cached]: 'None'
> [ECC:DEBUG]:[tools.py]:[search]:[Thread-9]: searching 'CMakeLists.txt' from 'C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo' to 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots'
> [ECC:DEBUG]:[tools.py]:[search]:[Thread-9]: found 'CMakeLists.txt' file: C:\src\master\PowerSuite\Android\VKGameHost\CMakeLists.txt
> [ECC:DEBUG]:[tools.py]:[contains]:[Thread-9]: found needed line: 'project(VKGameHost)'
> [ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-9]: [cmake]:[current]: 'C:\src\master\PowerSuite\Android\VKGameHost\CMakeLists.txt'
> [ECC:DEBUG]:[catkinizer.py]:[catkinize_if_needed]:[Thread-9]: Auto-catkinize supports only Linux currently.
> [ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-9]: [cmake]:[generate new db]
> [ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-9]: prefix paths: []
> [ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-9]: merged paths: 
> [ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-9]: CMAKE_PREFIX_PATH: 
> [ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-9]:  running command: ['cmake', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost']
> [ECC:DEBUG]:[tools.py]:[run_command]:[Thread-9]: executable file not found executing: ['cmake', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost']
> [ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-9]: cmake produced output: 
> None
> [ECC:ERROR]:[cmake_file.py]:[__compile_cmake]:[Thread-9]: cmake has finished, but no compilation database.
> [ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-9]: [db]:[get]: for file c:\src\master\powersuite\android\vkgamehost\jni\pokerbingo\pbscreen.cpp
> [ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-9]: [db]:[cached]: 'None'
> [ECC:DEBUG]:[tools.py]:[search]:[Thread-9]: searching 'compile_commands.json' from 'C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo' to 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots'
> [ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-9]: [CppProperties]:[get]: for file C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo\PBScreen.cpp
> [ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-9]: [CppProperties]:[cached]: 'None'
> [ECC:DEBUG]:[tools.py]:[search]:[Thread-9]: searching 'CppProperties.json' from 'C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo' to 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots'
> [ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-9]: [c_cpp_properties]:[get]: for file C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo\PBScreen.cpp
> [ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-9]: [c_cpp_properties]:[cached]: 'None'
> [ECC:DEBUG]:[tools.py]:[search]:[Thread-9]: searching 'c_cpp_properties.json' from 'C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo' to 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots'
> [ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-9]: [clang_complete_file]:[get]: for file C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo\PBScreen.cpp
> [ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-9]: [clang_complete_file]:[cached]: 'None'
> [ECC:DEBUG]:[tools.py]:[search]:[Thread-9]: searching '.clang_complete' from 'C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo' to 'C:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots'
> [ECC:DEBUG]:[bin_complete.py]:[run_clang_command]:[Thread-9]: clang command: 
> ['C:\\AndroidTools\\android-ndk-r18b\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\clang++.exe', '-c', '-fsyntax-only', '-x', 'c++', '-std=c++11', '-IC:\\AndroidTools\\android-ndk-r18b\\toolchains\\llvm\\prebuilt\\windows-x86_64\\lib64\\clang\\7.0.2\\include', '-I\\usr\\include', '-IC:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\Slots\\src', '-I\\usr\\lib\\clang\\7.0.2\\include', '-IC:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni\\PokerBingo', '-IC:\\src\\master\\PowerSuite\\Android\\VKGameHost\\jni', 'c:\\users\\sfield\\appdata\\local\\temp\\EasyClangComplete\\PBScreen.cpp']
> [ECC:DEBUG]:[tools.py]:[run_command]:[Thread-9]: command finished with code: 1
> [ECC:DEBUG]:[tools.py]:[run_command]:[Thread-9]: command output: 
> In file included from c:\users\sfield\appdata\local\temp\EasyClangComplete\PBScreen.cpp:1:
> 
> In file included from C:\src\master\PowerSuite\Android\VKGameHost\jni\PokerBingo/PBScreen.h:6:
> 
> C:\src\master\PowerSuite\Android\VKGameHost\jni\UI/Screen.h:7:10: fatal error: 'map' file not found
> 
> #include <map>
> 
>          ^~~~~
> 
> 1 error generated.
> 
> 
> [ECC:DEBUG]:[bin_complete.py]:[update]:[Thread-9]: rebuilding done in 2.034630060195923 seconds
> [ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-9]: generating error regions for view 106
> [ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-9]: 1 error regions ready
> [ECC:DEBUG]:[popup_error_vis.py]:[show_errors]:[Thread-9]: showing error regions: []
> [ECC:DEBUG]:[EasyClangComplete.py]:[config_updated]:[Thread-9]: updated config: <EasyClangComplete.plugin.view_config.ViewConfig object at 0x000001C877E5D9B0>
> [ECC:DEBUG]:[view_config.py]:[__remove_old_configs]:[Thread-2]: Skip young config: Age 13.719293117523193 < 1800. View: 106.
> [ECC:DEBUG]:[view_config.py]:[__remove_old_configs]:[Thread-20]: Skip young config: Age 43.7299370765686 < 1800. View: 106.
> [ECC:DEBUG]:[view_config.py]:[__remove_old_configs]:[Thread-21]: Skip young config: Age 73.74541997909546 < 1800. View: 106.
> [ECC:DEBUG]:[view_config.py]:[__remove_old_configs]:[Thread-22]: Skip young config: Age 103.74896001815796 < 1800. View: 106.
> [ECC:DEBUG]:[view_config.py]:[__remove_old_configs]:[Thread-23]: Skip young config: Age 133.75306797027588 < 1800. View: 106.
> [ECC:DEBUG]:[view_config.py]:[__remove_old_configs]:[Thread-24]: Skip young config: Age 163.75401711463928 < 1800. View: 106.

--- a/docs/index.md
+++ b/docs/index.md
@@ -24,6 +24,9 @@
 - **Other Systems** : use your package manager or install from clang website.
 - clang website: http://llvm.org/releases/download.html
 
+!!! tip
+    Make sure `clang` is available in your `PATH` or that the setting [clang binary](settings/#clang_binary) is set to the full path to your `clang` binary.
+
 ### **3. Configure your compiler flags and include folders**
 
 #### CMake and .sublime-project
@@ -33,7 +36,10 @@
 
 You're in luck! The plugin will run cmake on a proper `CMakeLists.txt` in your
 project folder and will use information from it to complete your code out of
-the box! For more details, read [here](../configs/#using-cmake-recommended).
+the box! For more details, read [here](configs/#using-cmake-recommended).
+
+!!! tip
+    Make sure `cmake` is available in your `PATH` or that the setting [cmake binary](settings/#cmake_binary) is set to the full path to your `clang` binary.
 
 #### Other options
 If you cannot use `CMake` in your project there are multiple ways to configure

--- a/docs/pc_readme.md
+++ b/docs/pc_readme.md
@@ -29,7 +29,7 @@ Don't worry! There are plenty of ways to configure the plugin! Read the related
 documentation [page](https://niosus.github.io/EasyClangComplete/configs/) for
 more info!
 
-## Extensive documentation
+## [Extensive documentation](https://niosus.github.io/EasyClangComplete/)
 There are so many things I want to tell you! There is so much the plugin is
 capable of! Read the [docs](https://niosus.github.io/EasyClangComplete/) to get
 started!

--- a/messages/install.rst
+++ b/messages/install.rst
@@ -7,7 +7,7 @@ A plugin for easy to use clang-based completions::
 Let't get started!
 ==================
 
-You're just two simple steps away!
+You're just two steps away!
 
 1. Install clang
 ----------------
@@ -27,65 +27,22 @@ Using CMake?
 Plugin will run cmake on a proper ``CMakeLists.txt`` in your project folder and
 will use information from it to complete your code out of the box.
 
-Have a compilation database?
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Plugin will search for a compilation database ``compile_commands.json`` in the
-project folder and will load it to complete your code. If you want to specify a
-custom path to a comilation database you can do it in settings::
-
-    "flags_sources": [
-        {"file": "compile_commands.json", "search_in": "<YOUR_PATH>"},
-    ]
-
-None of the above?
+Don't like CMake?
 ~~~~~~~~~~~~~~~~~~
 
-You will need a little bit of manual setup for now. Clang will automatically
-search for headers in the folder that contains the file you are working on and
-its parent. If you have a more sophisticated project you will need to help clang
-just a little bit. There are three ways to do it.
-
-Pick **ANY** of the following:
-
-- Set include dirs in ``"common_flags"`` setting in ``User Settings``.
-- Override ``"common_flags"`` setting in your project file, i.e. one that has
-  extension: ``*.sublime-project``. Just define the same setting in project
-  specific settings with either one of two prefixes: ``"ecc_"`` or
-  ``"easy_clang_complete_"`` to override a corresponding setting in your user
-  settings. See the project file in this repo for a working example. Minimal
-  example for clarity::
-
-    {
-      "settings":
-      {
-        "ecc_common_flags": ["-Isrc", "-I/usr/include"],
-        "easy_clang_complete_verbose": true
-      }
-    }
-
-- It is recommended to use one of the above, but if you already have a file with
-  flags, you can add ``.clang_complete`` file to the root of your project
-  folder. This file adds additional flags to the ones defined with in
-  ``"common_flags"``. Example::
-
-    -Isrc
-    -I/usr/include
+There are many ways of configuring the plugin!
+Read the docs here: https://niosus.github.io/EasyClangComplete/configs/
 
 That's it! You're ready to use the plugin!
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-More info here
-~~~~~~~~~~~~~~
+Intro and Documentation
+~~~~~~~~~~~~~~~~~~~~~~~
 
-Get more info in a readme:
-https://github.com/niosus/EasyClangComplete/blob/master/README.md
-
-Please see the default settings ``EasyClangComplete.sublime-settings``
-shipped with the plugin for explanations and sane default values.
+https://niosus.github.io/EasyClangComplete/
 
 Thanks!
 =======
 
 💜 this plugin? Consider buying me a 🍵
-https://github.com/niosus/EasyClangComplete#support-it
+https://niosus.github.io/EasyClangComplete/support/

